commands/cleanupwatchdog.c

```diff
@@ -358,6 +359,14 @@ main(
 
     if ((fp = fopen(WATCHDOG_INSTANCE_ID_FILE, "r")) == NULL)
     {
+        if (argc == 2 && !strcmp(argv[1], "--all"))
+        {
+            // watchdog slots are 1 and 2; 0 is for requesting a new slot
+            for (idindex = 1; idindex < 3; idindex++)
```

Comment:
There is a MAX_WATCHDOG_INSTANCE define, should we attempt to reset them all?

Reply:
That variable defines the maximum amount of watchdog threads allowed to be running (8). The number of slots in Xen is defined as [`NR_DOMAIN_WATCHDOG_TIMERS`](https://xenbits.xen.org/gitweb/?p=xen.git;a=blob;f=xen/include/xen/sched.h;h=edee52dfe41e8b23a35aa992ca021706fdabb6fc;hb=997d6248a9ae932d0dbaac8d8755c2b15fec25dc#l446) on Xen. I don't think it's exposed publicly.

Reply:
Given that you don't know many watchdogs Xen implements, and currently the use of {id = 0, timeout = 0} will instantly crash the host, would it help if that became "stop all watchdogs" ?

Reply:
Yep, that'd be much better. Since this fix is needed for Xenserver 7.1 CU2, the behaviour change would have to be included too.